server/src/internalClusterTest/java/org/elasticsearch/get/ShardMultiGetFomTranslogActionIT.java

```diff
@@ -93,19 +92,22 @@ public void testShardMultiGetFromTranslog() throws Exception {
         }
         assertThat(response.segmentGeneration(), equalTo(-1L));
         // Get followed by a Delete should still return a result
-        client().prepareDelete("test", "1").get();
-        response = getFromTranslog(indexOrAlias(), List.of("1", "2"));
+        var idToDelete = randomFrom(idsToIndex);
+        client().prepareDelete("test", idToDelete).get();
+        response = getFromTranslog(indexOrAlias(), idsToIndex);
         multiGetShardResponse = response.multiGetShardResponse();
         assertThat(getLocations(multiGetShardResponse).size(), equalTo(2));
         assertTrue(getFailures(multiGetShardResponse).stream().allMatch(Objects::isNull));
         getResponses = getResponses(multiGetShardResponse);
         assertThat(getResponses.size(), equalTo(2));
-        assertNotNull("get followed by a delete should still return a result", getResponses.get(0));
-        assertThat(getResponses.get(0).getId(), equalTo("1"));
-        assertThat(getResponses.get(0).isExists(), equalTo(false));
-        assertNotNull(getResponses.get(1));
-        assertThat(getResponses.get(1).getId(), equalTo("2"));
-        assertThat(getResponses.get(1).isExists(), equalTo(true));
+        var getResponseForDeletedId = getResponses.stream().filter(r -> r.getId().equals(idToDelete)).toList().get(0);
+        var getResponseForExistingId = getResponses.stream().filter(r -> r.getId().equals(idToDelete) == false).toList().get(0);
+        assertNotNull("get followed by a delete should still return a result", getResponseForDeletedId);
```

Comment:
I think this would never be null:
* if the stream is empty then `.get(0)` will throw no such element
* if the stream contains the null then `.toList()` will throw NPE when constructing.

I wonder if that should be something like `findFirst` or collecting the entire list and asserting its size is at least 1?

Reply:
`getResponses` could have `null` in it. For each ID we get one item in the list, but that item could be null. But you are right that the way I've changed the code already makes that check unnecessary. I've simplified it.